src/compile/data/data.ts

```diff
@@ -79,61 +83,151 @@ export interface SummaryComponent {
 // TODO: split this file into multiple files and remove this linter flag
 /* tslint:disable:no-use-before-declare */
 
-export function parseUnitData(model: UnitModel): DataComponent {
+/**
+ * Parses a model with data.
+ */
+function parseData(model: Model): DataComponent {
   return {
+    source: source.parseUnit(model),
     formatParse: formatParse.parseUnit(model),
-    nullFilter: nullFilter.parseUnit(model),
+    calculate: calculate.parseUnit(model),
     filter: filter.parseUnit(model),
-    nonPositiveFilter: nonPositiveFilter.parseUnit(model),
-
-    source: source.parseUnit(model),
     bin: bin.parseUnit(model),
-    calculate: formula.parseUnit(model),
     timeUnit: timeUnit.parseUnit(model),
-    timeUnitDomain: timeUnitDomain.parseUnit(model),
-    summary: summary.parseUnit(model),
+    aggregate: aggregate.parseUnit(model),
+    colorRank: colorRank.parseUnit(model),
     stackScale: stackScale.parseUnit(model),
-    colorRank: colorRank.parseUnit(model)
+    nonPositiveFilter: nonPositiveFilter.parseUnit(model),
+    nullFilter: nullFilter.parseUnit(model),
+    timeUnitDomain: timeUnitDomain.parseUnit(model)
   };
 }
 
+/**
+ * Create components for data source and transforms/ aggregates.
+ */
+export function parseUnitData(model: UnitModel): DataComponent {
+  return parseData(model);
+}
+
+/**
+ * The data component for a facet is the data component of its child.
```

Comment:
I don't think this is true.  

Facet can have its own `bin`, `timeUnit`, `calculate`, `filter`.